rust-src/concordium_base/src/protocol_level_tokens/token_module_state.rs

```diff
@@ -0,0 +1,81 @@
+use crate::{
+    common::{
+        cbor,
+        cbor::{Bytes, CborSerializationResult, SerializationOptions, UnknownMapKeys},
+    },
+    protocol_level_tokens::RawCbor,
+};
+
+use concordium_base_derive::{CborDeserialize, CborSerialize};
+
+/// Protocol level token (PLT) module state
+#[derive(Debug, Clone, Eq, PartialEq, CborSerialize, CborDeserialize)]
+pub struct TokenModuleState {
+    /// The name of the token
+    pub name:       String,
+    // /// A URL pointing to the token metadata
+    // TODO comment in as part of COR-1385
+    // pub metadata:   MetadataUrl,
+    /// Whether the token supports an allow list.
+    pub allow_list: Option<bool>,
+    /// Whether the token supports a deny list.
+    pub deny_list:  Option<bool>,
+    /// Whether the token is mintable.
+    pub mintable:   Option<bool>,
+    /// Whether the token is burnable.
+    pub burnable:   Option<bool>,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, CborSerialize, CborDeserialize)]
+pub struct MetadataUrl {
+    /// A string field representing the URL
+    pub url:              Option<String>,
```

Comment:
I don't think this should be optional - at least this is not how it's represented in the [cddl](https://github.com/Concordium/concordium-base/blob/fc2e84a081319d0ce0220b65543dcdf761c6226c/cbor-schema/plt.cddl#L165C1-L172C2)